### PR TITLE
Procedural skyboxes

### PR DESCRIPTION
**Pull request by jherico**
Because skyboxes are rendered from the Model library, I had to rearrange a bit of code to make a dependency structure where both skyboxes and entity renderers had access to the procedural structures.  Additionally I moved the network caching tools for both textures and shaders into a new gpu-networking library.

This also places more responsibility for consuming procedural shaders on the entity.  In this model, rather than having a special procedural fragment shader that it intended to work for all procedural entities, you need to customize the fragment shader you would *normally* use with some `#ifdefs`.  I think this approach will be more scalable, even if it requires a little more work from the consumers.  

You can see the potential of this functionality by coming to Dreaming and checking out the skybox at the default location

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5dcefde43ce77ae62f93d8c330cfaba1b288ba05 available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3869a441be8eef73d1ea6bbec6e232a20cbd903e available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

File: examples/shaders/exampleSkybox.fs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+// http://www.youtube.com/watch?v=1uFAu65tZpo
+
+//#define GREEN_VERSION
+
+// ** improved camera shaking
+// ** cleaned up code
+// ** added stuff to the gates
+
+// *******************************************************************************************
+// Please do NOT use this shader in your own productions/videos/games without my permission!
```
Comment (ZappoMan) [inline examples/shaders/exampleSkybox.fs]: did you get his permission?

Comment (jherico) [inline examples/shaders/exampleSkybox.fs]: I was just about to email him.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 712d93c1dd36b75427643f89dac2735c43c50e20 available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 72092b062ad682f2e3c6f040491ed16f34438859 available at: https://gist.github.com/8a010bb4853f563f4d26<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 65e3c9860463c74c85f7a9b80033a917134f17ec available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Comment by jherico** [comment]
test this please

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 65e3c9860463c74c85f7a9b80033a917134f17ec available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5f1affaff94454b4a79bf3d69f0ff070b88c0468 available at: https://gist.github.com/b0dceb7c6116f34a8f11<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5732.dmg

**Pull request merged**